move-vm-backend/src/deposit.rs

```diff
@@ -0,0 +1,49 @@
+#[cfg(not(feature = "std"))]
+use alloc::{borrow::ToOwned, vec, vec::Vec};
+use lazy_static::lazy_static;
+use move_core_types::{account_address::AccountAddress, ident_str, language_storage::StructTag};
+use serde::{Deserialize, Serialize};
+
+/// Mirroring structure of Move::deposit::Deposit
+/// Designed for bridging move transfer and native tokens of substrate
```

Comment:
related to this comment above

> I wouldn't mind even if we move the Deposit module to the stdlib module directly

if we decide to merge the Deposit module to move-stdlib, then we should move this file there as well since all this data here is tightly related to the actual Deposit Move module

Reply:
it's an extension and it might (and will be) changed multiple times. updating just this part and not entire STD will be cheaper from extrinsic call costs as well as building, injecting, etc.
we can put some todos to discuss, but I dont think this is critical at this stage of the project.

Reply:
> updating just this part and not entire STD will be cheaper

A user can always explicitly publish one module and not the whole bundle (the whole stdlib), so that's not a big concern.

> it's an extension and it might

But yes, that's a valid point! Let's pack it into another package (which could contain more extension packages/modules) and that way we'll decouple it from `move-vm-backend` (although the SubstrateAPI logic still depends on the correct `DEPOSIT_TEMPLATE`, but that can be provided by that new package library.

Reply:
that is not under user's control.
If we allow users to use their own published modules to do native assets transfer - how that would be considered a secure financial system?